### DISCUSSION TITLE
[11] Dinder - Clean Code Refactoring

### DISCUSSION THREAD

**Discussion by ghost** [dhbw-ka-tinf22b5-dinder/Dinder#16] [Projektblog]
# Clean Code Refactoring

### Frontend
- Mostly using biome.js, it caught errors in:
  - mostly imports
    - sorting by alphabet
    - declare TypeScript class imports with "type"
  - formatting
  - redundancy
    - e.g. semicolons, which are mostly optional in JavaScript, which is the cause for no other checker marking this as an error
  - too many or missing spaces in nested functions, making it unclear in which level of nesting the code is supposed to be in
  - `'` being used instead of `"`

### Backend
- Usage of Spring Profiles
- In-Memory Database (H2) for local tests, depending on the profile
    - now, an internet connection is optional to run the application
    - testing out changes without affecting the production database
- Database Connection configuration was changed from a custom bean to using the Spring autoconfiguration
- Flyway is also now loaded using the autoconfiguration
- The `AuthorityFactory` was renamed to `AuthoritySupplier` to better reflect the purpose of the interface

### Backend
- Research and testing for storing images in Supabase Storage
- Some endpoints were added, for example for receiving information about advertisements
- Added the `-parameters` flag to the compiler, because it was missing it caused parameter names to be missing in the compiled program, meaning spring couldn't parameter names through reflection
### Frontend
- http-client was updated to fetch advertisements

**Comment by Fefke** [comment]
Pointierter Blog-Beitrag,
ich finde gut, dass ihr mit biome.js eine statische Analyse auf euer Front-End gefahren 
und die Code-Base danach "standardisiert" habt.

Habt ihr noch vor das bei eurem Spring-Backend zu machen oder gibts da schon was integriertes? - _weiß ich nich. ^^_

**Comment by niklaskrauth** [comment]
Euer Refactoring im Frontend und Backend sieht sehr gut aus. Auch die Changes die ihr gemacht habt sehen sehr gut aus!

**Comment by Puggingtons** [comment]
Biome.js kannte ich bisher noch nicht, sieht aber sehr interessant aus. Das verbindet ja quasi alle Tools, die man für die Codequalität benutzt :D
Euer Fortschritt sieht soweit auch gut aus. Weiter so :)
